mlrun/projects/project.py

```diff
@@ -1857,13 +1878,15 @@ def set_function(
                 func, self
             )
             func["name"] = resolved_function_name
+
         elif hasattr(func, "to_dict"):
+            func = func.copy()
```

Comment:
Why are you copying the function? This changes existing flow in the sense that using `set_function(func,...)` would return the same `func` object and would also put that object in the project's dictionary. This allows the user to perform changes on `func` directly and these will apply to the function object as well.
With this change I must do `new_func = set_function(func,...)` and work on `new_func` instead. This breaks existing code flows so this needs to be reverted - if this is a problem then we need to discuss this.

Reply:
My thinking was that previously we were not changing the function so we should not change it now either. But I see your point that it changes the meaning of the original function. reveted.